coalib/output/Logging.py

```diff
@@ -1,28 +1,54 @@
 from datetime import datetime
+from collections import defaultdict
 import json
 import io
 import logging
 import logging.config
 
 
+class CounterHandler(logging.Handler):
+    """
+    A logging handler which counts the number of calls
+    for each logging level.
+    """
+    level_called = None
```

Comment:
Have you tried using a `collections.Counter` here?

And add a custom 'get' and 'set' method so that `level_called` can be private?